### PR TITLE
Added AWSUniqueId identification

### PR DESCRIPTION
**Pull request by lganzzzo**

### CODE REVIEW

File: signalfx-codahale/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java [hunk 1]
```diff
@@ -11,17 +12,22 @@
...
 import com.google.common.base.Optional;
 import com.google.common.collect.ImmutableSet;
+import com.signalfx.metrics.AWSInstanceInfo;
 import com.signalfx.metrics.SignalFxMetricsException;
 import com.signalfx.metrics.flush.AggregateMetricSender;
 import com.signalfx.metrics.protobuf.SignalFxProtocolBuffers;
 
 class AggregateMetricSenderSessionWrapper implements Closeable {
+	
+	private static final String AWS_UNIQUE_ID_DIMENSION_NAME = "AWSUniqueId";
```
Comment (tedoc2000) [inline signalfx-codahale/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java]: Indent

File: signalfx-yammer-example/pom.xml [hunk 2]
```diff
@@ -12,6 +12,8 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (tedoc2000) [inline signalfx-yammer-example/pom.xml]: Nope.. Can't use java 1.8

File: signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java [hunk 3]
```diff
@@ -26,12 +26,16 @@
  */
 
 class AggregateMetricSenderSessionWrapper implements Closeable {
+	
+	private static final String AWS_UNIQUE_ID_DIMENSION_NAME = "AWSUniqueId";
```
Comment (tedoc2000) [inline signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java]: Indent

**Comment by lganzzzo** [comment]
@tedoc2000  please review

File: pom.xml [hunk 4]
```diff
@@ -227,6 +227,11 @@
         <artifactId>commons-io</artifactId>
         <version>2.4</version>
       </dependency>
+      <dependency>
```
Comment (tedoc2000) [inline pom.xml]: Find another json parser.. this one is compiled for java 8. We need one that is compiled for java 6

**Comment by mpetazzoni** [comment]
Also, indent is wrong and files are missing copyright headers. This is public code, we have to be careful about this.

**Comment by lganzzzo** [comment]
@tedoc2000 please review

**Comment by lganzzzo** [comment]
@tedoc2000 @mpetazzoni please review

File: signalfx-yammer-example/pom.xml [hunk 5]
```diff
@@ -9,17 +9,13 @@
 
   <name>signalfx.yammer.example</name>
   <url>http://maven.apache.org</url>
-
```
Comment (tedoc2000) [inline signalfx-yammer-example/pom.xml]: Why did you remove this?

File: signalfx-java/src/main/java/com/signalfx/metrics/aws/AWSInstanceInfo.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import org.apache.http.util.EntityUtils;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AWSInstanceInfo {
+
+    public static final String AWS_UNIQUE_ID_DIMENSION_NAME = "AWSUniqueId";
+
+    private static final String URL = "http://169.254.169.254/2014-11-05/dynamic/instance-identity/document";
```
Comment (tedoc2000) [inline signalfx-java/src/main/java/com/signalfx/metrics/aws/AWSInstanceInfo.java]: Turns out 2014-11-05 doesn't work on some systems.. Lets use latest

File: signalfx-yammer-example/pom.xml (hunk 5 above)
Comment (lganzzzo) [inline signalfx-yammer-example/pom.xml]: Fixed

**Comment by lganzzzo** [comment]
Changed to .../latest/...
@tedoc2000 @mpetazzoni  please review

**Comment by tedoc2000** [comment]
@lganzzzo can you rebase this so that we only have one commit.. To look at instead of the merge commit and the actually code you changed. Thanks!

**Comment by lganzzzo** [comment]
@tedoc2000 rebased, please review
(If I correctly understood what you meant:) )

**Comment by mpetazzoni** [comment]
This is superseded by #15.
